Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs

```diff
@@ -0,0 +1,233 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Audio;
+using Microsoft.MixedReality.Toolkit.SpatialAwareness;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Examples
+{
+    /// <summary>
+    /// Demonstration class using WindowsMicrophoneStream (from com.microsoft.mixedreality.toolkit.micstream) to select the 
+    /// voice microphone and adjust the spatial awareness mesh based on the amplitude of the user's voice.
+    /// </summary>
+    [RequireComponent(typeof(AudioSource))]
+    public class MicrophoneAmplitudeDemo : MonoBehaviour
+    {
+#if MICSTREAM_PRESENT
+
+        [SerializeField]
+        [Tooltip("Gain to apply to the microphone input.")]
+        [Range(0, 10)]
+        private float inputGain = 1.0f;
+
+        [SerializeField]
+        [Tooltip("Factor by which to boost the microphone amplitude when changing the mesh display.")]
+        [Range(0, 50)]
+        private int amplitudeBoostFactor = 10;
+
+        private IMixedRealitySpatialAwarenessSystem spatialAwarenessSystem = null;
+
+        /// <summary>
+        /// Instance of the spatial awareness system.
+        /// </summary>
+        private IMixedRealitySpatialAwarenessSystem SpatialAwarenessSystem
+        {
+            get
+            {
+                if (spatialAwarenessSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealitySpatialAwarenessSystem>(out spatialAwarenessSystem);
+                }
+                return spatialAwarenessSystem;
+            }
+        }
+
+        private IMixedRealitySpatialAwarenessMeshObserver spatialMeshObserver = null;
+
+        /// <summary>
+        /// Instance of a spatial awareness mesh observer. Used to acquire the visible mesh material.
+        /// </summary>
+        private IMixedRealitySpatialAwarenessMeshObserver SpatialMeshObserver
+        {
+            get
+            {
+                if (spatialMeshObserver == null)
+                {
+                    spatialMeshObserver = (SpatialAwarenessSystem as IMixedRealityDataProviderAccess)?.GetDataProvider<IMixedRealitySpatialAwarenessMeshObserver>();
+                }
+                return spatialMeshObserver;
+            }
+        }
+
+        private Material visibleMaterial = null;
+
+        /// <summary>
+        /// The visible mesh material in use by the spatial mesh observer.
+        /// </summary>
+        private Material VisibleMaterial
+        {
+            get
+            {
+                if (visibleMaterial == null)
```

Comment:
Is VisibleMaterial meant to be private or public?

Reply:
same as the observer. thanks!

Reply:
removed the property in favor of a simple private field